dataset/bin.cpp

```diff
@@ -75,7 +75,8 @@ void update_indices_by_binning(const VariableView &indices,
 template <class Index>
 Variable groups_to_map(const VariableConstView &var, const Dim dim) {
   return variable::transform(subspan_view(var, dim),
-                             core::element::groups_to_map<Index>);
+                             core::element::groups_to_map<Index>,
+                             "groups_to_map");
```

Comment:
Since these are implementation details of `bin`, is it worth including this in the operator name, something like `bin.groups_to_map`?

Reply:
I would not use 'bin' because that makes it look a bit like a member of the Python class 'Bin'. Maybe somethinlg along the lines of '<internal>.groups_to_map'?

Reply:
Well, but the user will likely see this message when calling `sc.bin`, so maybe `scipp.bin.groups_to_map`?